### PR TITLE
Add vertical pod autoscaler metrics in KSM

### PR DESCRIPTION
**Pull request by QuentinBisson**
Towards https://github.com/giantswarm/giantswarm/issues/26221

This PR enables VPA metrics coming from KSM

### CODE REVIEW

File: helm/prometheus-operator-app/values.yaml [hunk 1]
```diff
@@ -193,6 +193,47 @@ prometheus-operator-app:
...
+            labelsFromPath:
+              verticalpodautoscaler: [metadata, name]
+              namespace: [metadata, namespace]
+              target_api_version: [apiVersion]
+              target_kind: [spec, targetRef, kind]
+              target_name: [spec, targetRef, name]
+            metricNamePrefix: kube_verticalpodautoscaler
+            metrics:
+              - name: "annotations"
+                help: "Kubernetes annotations converted to Prometheus labels."
```
Comment (hervenicol) [inline helm/prometheus-operator-app/values.yaml:218]: Annotations and labels can be high cardinality.
Is it something added here, or something that was in defaults and has to be explicitely set here to be kept?
Maybe there's a way to only keep a few specific labels/annotations we use?

**Review by hervenicol — COMMENTED** [review]
I'm fine with the VPA info, but not fond of the K8S labels and annotations set as Prometheus labels.

File: helm/prometheus-operator-app/values.yaml (hunk 1 above)
Comment (QuentinBisson) [reply helm/prometheus-operator-app/values.yaml:218]: So I tried Yesterday to look at how we should configure this by keeping the default metrics, especially those that could bé useful to us around the status values but labels and annotations are thé only one that worked ... It's a bit challenging

**Comment by QuentinBisson** [comment]
Blocking this waiting for a full upstream configuration https://github.com/kubernetes/kube-state-metrics/issues/2041

**Comment by QuentinBisson** [comment]
Moved to the old way of doing things (pre 2.9.0 until 2.9.0 is actually released and it's documented as per https://github.com/kubernetes/kube-state-metrics/issues/2041)

Tested and working:

image

**Comment by QuentinBisson** [comment]
@hervenicol do you want to review this? :)

**Review by hervenicol — APPROVED** [review]
LGTM

**Pull request merged**